src/MergeContext.js

```diff
@@ -973,8 +973,18 @@ class PullRequest {
                 statusChecks.addOptionalStatus(new StatusCheck(st));
         }
 
+        // Returns whole check runs history for the commit.
+        // We need only the freshest checks.
         const checkRuns = await GH.getCheckRuns(this._prHeadSha());
-        for (let st of checkRuns) {
+        // check.id is greater in newer checks
+        const sortedCheckRuns = checkRuns.sort((e1, e2) => parseInt(e1.id) - parseInt(e2.id));
+        let uniqueCheckRuns = [];
+        for (let i = sortedCheckRuns.length -1; i >= 0; --i) {
+            const check = sortedCheckRuns[i];
```

Comment:
Feels odd to sort the array one way and then work extra hard to iterate it in the reverse order. Please see whether you can adjust sortedCheckRuns so that freshest checks come first and then use forEach() to iterate that array. Something along these untested lines may work:

```suggestion
        const sortedCheckRuns = checkRuns.sort((e1, e2) => parseInt(e2.id) - parseInt(e1.id));
        let uniqueCheckRuns = [];
        sortedCheckRuns.forEach(check => {
```